### PR TITLE
Refactor handlers

### PR DESCRIPTION
**Pull request by GabrielFleischer**
Refactor handlers so that they use java exceptions to handle exceptional behavior.

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/WitnessMessageHandler.java [hunk 1]
```diff
@@ -31,14 +35,14 @@ private WitnessMessageHandler() {
    * @param laoRepository the repository to access the LAO of the channel
    * @param channel the channel on which the message was received
    * @param senderPk the public key of the sender
-   * @param message the message that was received
+   * @param data the message that was received
    * @return true if the message cannot be processed and false otherwise
    */
   public static boolean handleWitnessMessage(
```
Comment (arnauds5) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/WitnessMessageHandler.java:41]: The handleWitnessMessage is still using a boolean return value.

Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/WitnessMessageHandler.java]: oops, fixing it

**Review by arnauds5 — DISMISSED** [review]

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/GenericHandler.java [hunk 2]
```diff
@@ -107,11 +109,14 @@ public static void handleCatchup(
...
+    Log.d(TAG, "Messages to handle : " + messages.size());
     // Handle all received messages from the catchup
     for (MessageGeneral msg : messages) {
-      boolean enqueue = handleMessage(laoRepository, channel, msg);
-      if (enqueue) {
+      try {
+        handleMessage(laoRepository, channel, msg);
+      } catch (DataHandlingException ex) {
+        Log.e(TAG, "Unable to handle message", ex);
+        // This will have to change at some point
```
Comment (jbsv) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/GenericHandler.java:119]: Probably true for any line of code in any program ;-)
Can we have more insight why you think so ?

**Review by Lugiasc — APPROVED** [review]
Aside from Jean's comment, seems good to me !

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/GenericHandler.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/GenericHandler.java]: I updated it

**Review by Lugiasc — DISMISSED** [review]

**Review by arnauds5 — APPROVED** [review]

**Review by jbsv — APPROVED** [review]

**Pull request merged**